workflow/controller/operator.go

```diff
@@ -2103,6 +2117,10 @@ func (woc *wfOperationCtx) markNodePhase(nodeName string, phase wfv1.NodePhase,
 }
 
 func (woc *wfOperationCtx) onNodeComplete(node *wfv1.NodeStatus) {
+	woc.recordNodePhaseEvent(node)
+}
+
+func (woc *wfOperationCtx) recordNodePhaseEvent(node *wfv1.NodeStatus) {
```

Comment:
could we just invoke this in `woc.markNodePhase()`?

Reply:
Yes, it is invoked there on lines 2093 to 2095. Did you have something else in mind?